WooCommerce/Woo Watch App/Orders/OrderDetailLoader.swift

```diff
@@ -21,6 +26,9 @@ struct OrderDetailLoader: View {
                     .redacted(reason: .placeholder)
             case .loaded(let order):
                 OrderDetailView(order: order)
+                    .onAppear {
```

Comment:
I was thinking if I should use `.onAppear` here to call `MarkOrderAsReadUseCase.markOrderNoteAsReadIfNeeded` or call it directly in `OrderDetailView`, but I went with this just so we do this marking logic when we do the order fetching from the push notification and not always when we show the order from the watch app.
What do you think?

Reply:
I think a better place to not add extra logic to the UI and keep it containable in the VM would be:

- Adda a `markNotification...` method in the `OrderNotificationDataService` which will call your use case.
- Add a `markNotification...` method in the `OrderDetailLoaderViewModel` and call the method from the data service
- Call that VM method from the `OrderDetailLoaderView`, it could be on a `task` modifier so it not called multiple times (possibly)

What do you think?

Reply:
> * Call that VM method from the `OrderDetailLoaderView`

you mean `OrderDetailView`?

Reply:
Since `OrderDetailView` is used both in `OrderDetailLoader` and `OrdersListView` I was thinking of just having that marking in `OrderDetailLoader` since there we do the loading related to the push notification.

Also I wanted to avoid adding marking logic in `OrderNotificationDataService` since that is used for load/fetching objects and not presenting the order, and we need to mark the notification read if the order is shown, not just loaded/fetched.

I would maybe just change this to:
```
OrderDetailView(order: order)
          .task {
            await MarkOrderAsReadUseCase.markOrderNoteAsReadIfNeeded(network: self.network,
                                         noteID: pushNotification.noteID,
                                         orderID: Int(order.orderID))
          }
```

In short:
- in the app we will have marking in `OrderLoaderViewController` (`markNotificationAsReadIfNeeded(note: Note)`) which is called if users just tap the notification or enter the app via notification extension.
- on watch we have marking in `OrderDetailLoader` when we present `OrderDetailView` with loaded order

Reply:
I understand! Will try to update it now 👍